internal/gps/maybe_source.go

```diff
@@ -47,7 +46,11 @@ func (mbs maybeSources) try(ctx context.Context, cachedir string, c singleSource
 		if err == nil {
 			return src, state, nil
 		}
-		errs = append(errs, errors.Wrapf(err, "failed to set up %q", mb.getURL()))
+		urls := ""
+		for _, url := range mb.possibleURLs() {
+			urls += url.String() + "\n"
+		}
+		errs = append(errs, errors.Wrapf(err, "failed to set up %q", urls))
```

Comment:
nit: let's make an incremental improvement on the output we currently have and change the message to `"failed to set sources from the following URLs:\n%s"`.

(note also, use `%s` instead of `%q`, as otherwise we'll end up quoting the whole group of URLs, which i don't think makes a ton of sense.)